### ISSUE TITLE
When mounting SD card (problem)

### ISSUE THREAD

**Issue by Jefke1** [Edzelf/ESP32-Radio#309]
Hi, radio plays very well, when i connect the sd card, i no longer get any sound from the radio and from the sd card . What am I doing wrong ??
Chip = VLSI 1053b

Log before:
D: Slow SPI, Testing VS1053 read/write registers...
D: Fast SPI, Testing VS1053 read/write registers again...
D: endFillByte is 0

Afther: With SD card

D: Starting ESP32-radio running on CPU 1 at 240 MHz.  Version Mon, 25 Mar 2020 10:45:00 GMT.  Free memory 294052
D: Display type is NEXTION
D: Partition nvs found, 20480 bytes
D: Read 43 keys from NVS
D: pin_ir set to 0
D: Pin 0 is already reserved!
D: pin_enc_clk set to 0
D: Pin 0 is already reserved!
D: pin_enc_dt set to 0
D: Pin 0 is already reserved!
D: pin_enc_sw set to 0
D: Pin 0 is already reserved!
D: pin_tft_cs set to 0
D: Pin 0 is already reserved!
D: pin_tft_dc set to 0
D: pin_tft_scl set to -1
D: pin_tft_sda set to -1
D: pin_tft_bl set to -1
D: pin_tft_blx set to -1
D: pin_sd_cs set to 21
D: pin_vs_cs set to 5
D: pin_vs_dcs set to 32
D: pin_vs_dreq set to 4
D: pin_shutdown set to -1
D: pin_shutdownx set to -1
D: pin_spi_sck set to 18
D: pin_spi_miso set to 19
D: pin_spi_mosi set to 23
D: GPIO0 is HIGH
D: GPIO2 is HIGH
D: GPIO4 is HIGH
D: GPIO5 is HIGH
D: GPIO12 is HIGH
D: GPIO13 is HIGH
D: GPIO14 is HIGH
D: GPIO15 is HIGH
D: GPIO16 is HIGH
D: GPIO17 is HIGH
D: GPIO18 is LOW, probably no PULL-UP
D: GPIO19 is HIGH
D: GPIO21 is HIGH
D: GPIO22 is HIGH
D: GPIO23 is HIGH
D: GPIO25 is HIGH
D: GPIO26 is HIGH
D: GPIO27 is HIGH
D: GPIO32 is HIGH
D: GPIO33 is HIGH
D: GPIO34 is LOW, probably no PULL-UP
D: GPIO35 is LOW, probably no PULL-UP
D: GPIO39 is LOW, probably no PULL-UP
D: gpio_12 will execute upvolume = 2
D: gpio_13 will execute downvolume = 2
D: Enable pin 0 for IR
D: Start display
D: Init Nextion, pins 16,17
D: Nextion command cls BLACK
D: Nextion command page 0
D: Locate mp3 files on SD, may take a while...
D: SD directory is /
D: SD directory is /System Volume Information
D: 10 tracks on SD
D: Create list with acceptable WiFi networks
D: Added Orange-486AE to list of networks
D: Added SSID2 to list of networks
D: End adding networks
D: Scan Networks
D: Scan completed
D: Number of available networks: 9
D:  1 - Orange-486AE              Signal: -72 dBm, Encryption WPA2_PSK, Acceptable
D:  2 - telenet-0ABC6             Signal: -79 dBm, Encryption WPA_WPA2_PSK, 
D:  3 - TelenetWiFree             Signal: -80 dBm, Encryption ????, 
D:  4 - Proximus Public Wi-Fi     Signal: -83 dBm, Encryption ????, 
D:  5 - TelenetWiFree             Signal: -92 dBm, Encryption ????, 
D:  6 - telenet-05670             Signal: -93 dBm, Encryption WPA_WPA2_PSK, 
D:  7 - telenet-30B52             Signal: -94 dBm, Encryption WPA_WPA2_PSK, 
D:  8 - WiFi-2.4-DCBC             Signal: -94 dBm, Encryption WPA2_PSK, 
D:  9 - TelenetWiFree             Signal: -94 dBm, Encryption ????, 
D: End of list
D: Command: clk_dst with parameter 1
D: Command: clk_offset with parameter 1
D: Command: clk_server with parameter pool.ntp.org
D: Command: gpio_00 with parameter uppreset = 1
D: Command: gpio_12 with parameter upvolume = 2
D: Command: gpio_13 with parameter downvolume = 2
D: Command: ir_40bf with parameter upvolume = 2
D: Command: ir_c03f with parameter downvolume = 2
D: Command: mqttbroker with parameter none
D: Command: mqttpasswd with parameter *******
D: Command: mqttport with parameter 1883
D: Command: mqttprefix with parameter none
D: Command: mqttuser with parameter none
D: Command: pin_enc_clk with parameter 0
D: Command: pin_enc_dt with parameter 0
D: Command: pin_enc_sw with parameter 0
D: Command: pin_ir with parameter 0
D: Command: pin_sd_cs with parameter 21
D: Command: pin_tft_cs with parameter 0
D: Command: pin_tft_dc with parameter 0
D: Command: pin_vs_cs with parameter 5
D: Command: pin_vs_dcs with parameter 32
D: Command: pin_vs_dreq with parameter 4
D: Command: preset with parameter 0
D: Command: preset_00 with parameter 109.206.96.34:8100
D: Command: preset_01 with parameter airspectrum.cdnstream1.com:8114/1648_128
D: Command: preset_02 with parameter us2.internet-radio.com:8050
D: Command: preset_03 with parameter airspectrum.cdnstream1.com:8000/1261_192
D: Command: preset_04 with parameter airspectrum.cdnstream1.com:8008/1604_128
D: Command: preset_05 with parameter us1.internet-radio.com:8105
D: Command: preset_06 with parameter icecast.omroep.nl:80/radio1-bb-mp3
D: Command: preset_07 with parameter 205.164.62.15:10032
D: Command: preset_08 with parameter skonto.ls.lv:8002/mp3
D: Command: preset_09 with parameter 94.23.66.155:8106
D: Command: preset_10 with parameter ihr/IHR_IEDM
D: Command: preset_11 with parameter ihr/IHR_TRAN
D: Command: toneha with parameter 13
D: Command: tonehf with parameter 1
D: Command: tonela with parameter 10
D: Command: tonelf with parameter 11
D: Command: volume with parameter 76
D: Slow SPI, Testing VS1053 read/write registers...
D: VS1053 SPI error. SB:0000 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:012C R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0258 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0384 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:04B0 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:05DC R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0708 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0834 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0960 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0A8C R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0BB8 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0CE4 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0E10 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:0F3C R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:1068 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:1194 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:12C0 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:13EC R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:1518 R1:FFFF R2:FFFF
D: VS1053 SPI error. SB:1644 R1:FFFF R2:FFFF
D: This is not a VS1053, but a VS1103 instead!
D: Connect to WiFi
D: Connected to Orange-486AE
D: IP = 192.168.0.32
D: Nextion command page 1
D: Start server for commands
D: Network found. Starting mqtt and OTA
D: MDNS responder started
D: Rotary encoder is disabled (0/0/0)
D: Sync TOD
D: Sync TOD, new value is 17:34:00
D: STOP requested
D: New preset/file requested (0/0) from 109.206.96.34:8100
D: Connect to new host 109.206.96.34:8100
D: Connect to 109.206.96.34 on port 8100, extension /
D: Connected to server
D: Duration mp3loop 1
D: SPI semaphore not taken within 270 ticks by CPU 1, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 280 ticks by CPU 1, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 290 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 300 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 310 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 320 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D: SPI semaphore not taken within 330 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D: Song stopped incorrectly!
D: REG   Contents
D: ---   -----
D: SPI semaphore not taken within 340 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D:   0 -  FFFF
D:   1 -  FFFF
D: SPI semaphore not taken within 350 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsongD:   2 -  FFFF

D:   3 -  FFFF
D:   4 -  FFFF
D: SPI semaphore not taken within 360 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D:   5 -  FFFF
D:   6 -  FFFF
D: SPI semaphore not taken within 370 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D:   7 -  FFFF
D:   8 -  FFFF
D: SPI semaphore not taken within 380 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsongD:   9 -  FFFF

D:   A -  FFFF
D:   B -  FFFF
D: SPI semaphore not taken within 390 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D:   C -  FFFF
D:   D -  FFFF
D: SPI semaphore not taken within 400 ticks by CPU 0, id hspec
D: Semaphore is claimed by stopsong
D:   E -  FFFF
D:   F -  FFFF
D: Nextion t0.txt="ESP32-Radio"
D: Switch to HEADER
D: Duration mp3loop 3
D: Headerline: icy-name:NAXI LOVE RADIO (NAXI,Belgrade,Serbia, NAXI,Beograd,Srbija) - 128k
D: Headerline: icy-genre:Love and Romance
D: Headerline: icy-url:http://www.naxi.rs
D: Headerline: content-type:audio/mpeg
D: audio/mpeg seen.
D: D: Nextion t2.txt="NAXI LOVE RADIO (NAXI,Belgrade,Serbia, NAXI,Beograd,Srbija) - 128k"Nextion t2.txt="NAXI LOVE RADIO (NAXI,Belgrade,Serbia, NAXI,Beograd,Srbija) - 128k"

D: Headerline: icy-metaint:8192
D: Headerline: icy-br:128
D: Switch to DATA, bitrate is 128, metaint is 8192
D: Duration mp3loop 40
D: Metadata block 64 bytes
D: Streamtitle found, 48 bytes
D: StreamTitle=' Mariah Carey - Hero';StreamUrl='';
D: Nextion t1.txt=" Mariah Carey\rHero"
D: NEXTION command seen 70 volume=75
D: Command: volume with parameter 75
D: Metadata block 80 bytes
D: Streamtitle found, 64 bytes
D: StreamTitle=' James Arthur - Say You Won't Let Go';StreamUrl='';
D: Nextion t1.txt=" James Arthur\rSay You Won't Let Go"

**Comment by Edzelf** [comment]
Some SD cards disturb the SPI bus.  Search for the solution in other issues.

**Comment by Jefke1** [comment]
Hi Ed, I mounted another SD card reader.
Playing MP3s, but artist and title are displayed for a fraction of a sec in t1 of the Nextion display. In the web interface is everything ok.
D: NEXTION command seen 70 uppreset=1
D: Command: uppreset with parameter 1
D: SD_currentnode is 0, curnod is 0, delta is 1
D: getSDfilename random choice
D: getSDfilename requested node ID is 2,2,0,0
D: Selected file is localhost/Mp3/Dido - Thank You (Acoustic).mp3
D: Playing localhost/Mp3/Dido - Thank You (Acoustic).mp3
D: STOP requested
D: Found ID3 info
D: Nextion t0.txt="ESP32 MP3 Player"
D: Nextion t1.txt=""
D: Nextion t2.txt="Playing from local file"
D: Song stopped correctly after 20 msec
D: NEXTION command seen 70 downpreset=1
D: Command: downpreset with parameter 1
D: SD_currentnode is 0, curnod is 0, delta is -1
D: getSDfilename random choice
D: getSDfilename requested node ID is 2,1,0,0
D: Selected file is localhost/Mp3/David Gilmour Wish You Were Here  1080p HD.mp3
D: Playing localhost/Mp3/David Gilmour Wish You Were Here  1080p HD.mp3
D: STOP requested
D: Found ID3 info
D: Nextion t0.txt="ESP32 MP3 Player"
D: Nextion t1.txt=""
D: Nextion t2.txt="Playing from local file"
D: Song stopped correctly after 20 msec


thanks in advance
